lib/internal/fs/streams.js

```diff
@@ -133,7 +132,6 @@ function importFd(stream, options) {
       stream.fd = options.fd.fd;
       stream[kFs] = FileHandleOperations(stream[kHandle]);
       stream[kHandle][kRef]();
-      options.fd.on('close', FunctionPrototypeBind(stream.close, stream));
```

Comment:
Hmm... the mixin approach here bothers me. Another way we could do this would be to follow the model of `AbortController/AbortSignal` ... 

```js
class CloseSignal extends EventTarget {
  // ...
  get closed() { /* ... * / }
}

class FileHandle extends JSTransferable {
  constructor(...) {
    // ...
    this[kCloseSignal] = new CloseSignal();
  }
  // ...
  get closeSignal() {
    return this[kCloseSignal];
  }
}

// then, in the File stream...

options.fd.closeSignal.onclose = FunctionPrototypeBind(stream.close, stream);
```

Reply:
Either way this still needs tests and doc updates.

Reply:
@jasnell it bothers me as well... but this PR removes it?

Reply:
Ha!!! I completely misread that! Sigh. It's a good thing it's the weekend

Reply:
@mmomtchev FYI

Reply:
also would be nice if there was a more descriptive error provided to the stream when this happens, i.e. instead of `stream.close` it would do something like `stream.destroy(new FileClosedError())`